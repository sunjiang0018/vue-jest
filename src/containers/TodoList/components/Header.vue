<template>
  <div class="header">
    <div class="header-contain">
      TodoList
      <input
        type="text"
        data-test="header-input"
        v-model="inputValue"
        placeholder="input TodoItem"
        @keyup.enter="addTodoItem"
      />
    </div>

  </div>
</template>

<script>
export default {
  name: 'Header',
  data () {
    return {
      inputValue: ''
    }
  },
  methods: {
    addTodoItem () {
      if (this.inputValue) {
        this.$emit('add', this.inputValue)
        this.inputValue = ''
      }
    }
  }
}
</script>

<style>

.header{
  height: 50px;
  width: 100%;
  background-color: #333;
}

.header-contain{
  width:  600px;
  margin: 0 auto;
  color: #fff;
  line-height: 50px;
  font-size: 24px;
  display: flex;
  justify-content: space-between;
}

.header-contain > input[type="text"]{
  width: 230px;
  margin: 5px 0;
  color: #333;
  text-indent: 10px;
  font-size: 1rem;
}
</style>
