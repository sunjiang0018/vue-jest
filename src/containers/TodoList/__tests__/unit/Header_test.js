import { shallowMount } from '@vue/test-utils'
import Header from '../../components/Header'
describe('header Test', () => {
  it('监测header组件样式是否发生变换', () => {
    const wrapper = shallowMount(Header)
    expect(wrapper).toMatchSnapshot()
  })
  it('header 有一个input', () => {
    const wrapper = shallowMount(Header)
    const input = wrapper.find("[data-test='input']")
    expect(input.exists()).toBeTruthy()
  })

  it('header 中input 内容为空', () => {
    const wrapper = shallowMount(Header)
    const inputValue = wrapper.vm.$data.inputValue
    expect(inputValue).toBe('')
  })

  it('header 中input 有内容时', () => {
    const wrapper = shallowMount(Header)
    const input = wrapper.find('[data-test="input"]')
    input.setValue('test')
    const inputValue = wrapper.vm.$data.inputValue
    expect(inputValue).toBe('test')
  })

  it('header 中input内容为空时，不发送add事件', () => {
    const wrapper = shallowMount(Header)
    const input = wrapper.find('[data-test="input"]')
    input.setValue('')
    input.trigger('keyup.enter')
    expect(wrapper.emitted().add).toBeFalsy()
  })

  it('header 中input内容不为空时，触发add事件', () => {
    const wrapper = shallowMount(Header)
    const input = wrapper.find('[data-test="input"]')
    input.setValue('test keyup.enter not empty')
    input.trigger('keyup.enter')
    expect(wrapper.emitted().add).toBeTruthy()
  })

  it('header 中input触发add事件后，清空内容', () => {
    const wrapper = shallowMount(Header)
    const input = wrapper.find('[data-test="input"]')
    input.setValue('test keyup.enter not empty')
    input.trigger('keyup.enter')
    expect(wrapper.emitted().add).toBeTruthy()
    const inputValue = wrapper.vm.$data.inputValue
    expect(inputValue).toBe('')
  })
})
