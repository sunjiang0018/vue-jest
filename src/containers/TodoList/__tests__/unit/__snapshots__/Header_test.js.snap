// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header Test 监测header组件样式是否发生变换 1`] = `
<div class="header">
  <div class="header-contain">
    TodoList
    <input type="text" data-test="header-input" placeholder="input TodoItem">
  </div>
</div>
`;
